src/openrct2/world/Map.cpp

```diff
@@ -1361,22 +1361,26 @@ std::unique_ptr<GameActions::ConstructClearResult> MapCanConstructWithClearAt(
             res->GroundFlags |= ELEMENT_IS_UNDERWATER;
             if (water_height < pos.clearanceZ)
             {
+                bool returnError = true;
```

Comment:
On commit `https://github.com/OpenRCT2/OpenRCT2/pull/15024/commits/41e564d16c73764fb8a567ed3765207e5f79255c`, seems like this would be better off as just `error` or something similar. Not a big deal though